tests/backup/backup_restore_basic_test.go

```diff
@@ -895,3 +896,197 @@ var _ = Describe("{CustomResourceRestore}", func() {
+			for _, namespace := range bkpNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					sem <- struct{}{}
+					time.Sleep(timeBetweenConsecutiveBackups)
+					backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+					backupNames = append(backupNames, backupName)
+					wg.Add(1)
+					go func(backupName string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						defer func() { <-sem }()
+						err = CreateBackup(backupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace},
+							labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation: %s", backupName))
+					}(backupName)
+				}
+				wg.Wait()
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Remove the backup location where backups were taken", func() {
+			log.InfoD("Remove backup location where backups were taken")
+			err := DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID, false)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", customBackupLocationName))
+			//TODO: Eliminate time.Sleep
+			time.Sleep(time.Minute * 3)
+		})
+
+		Step("Add the backup location again which had backups", func() {
+			log.InfoD("Add the backup location with backups back")
+			for _, provider := range providers {
+				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				err := CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
+			}
+		})
+		Step("Taking backup of applications to trigger BackupSync goroutine", func() {
+			log.InfoD("Taking backup of applications to trigger BackupSync goroutine")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx failed")
+			for _, namespace := range bkpNamespaces {
+				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				backupNamespaceMap[namespace] = backupName
+				err = CreateBackup(backupName, SourceClusterName, customBackupLocationName, backupLocationUID, []string{namespace}, labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation %s", backupName))
+				backupNames = append(backupNames, backupName)
+			}
+		})
+
+		Step("Check if all backups are synced or not", func() {
+			log.InfoD("Check if backups created before are synced or not")
+
+			// Wait for backups to get synced
+			for i := 0; i < 1000; i++ {
+				fetchedBackupNames, _ := GetAllBackupsAdmin()
+				log.Infof(fmt.Sprintf("Fetched backups %d", len(fetchedBackupNames)))
+				if len(fetchedBackupNames) == len(backupNames) {
+					break
+				} else {
+					time.Sleep(30)
+				}
+			}
```

Comment:
We can use the ready-made `task.DoRetryWithTimeout` function here

Reply:
I tried using it when I try to using this but the problem here is there is no way to pass external variables when you take this approach. Here I am mostly using backupNames which is picked up earlier and then using it  to compare with the newer fetched value. 
```
backupLocationDeleteStatusCheck := func() (interface{}, bool, error) {
```

Reply:
Here you can keep enumerating the backups and check until the length is what is expected. After that you can come out and proceed. Something like

```
backupCountCheck = func() (interface{}, bool, error) {
    fetchedBackupNames, _ := GetAllBackupsAdmin()
    og.Infof(fmt.Sprintf("Fetched backups %d", len(fetchedBackupNames)))
				if len(fetchedBackupNames) == len(backupNames) {
					return "", false, nil
				} else {
					return "", true, fmt.Errorf("Expected length was %d but got %d", len(backupNames), len(fetchedBackupNames))
				}
}

_, err := task.DoRetryWithTimeout(backupCountCheck, Timeout, RetryTime)

```

Reply:
Done

Reply:
I tried to see if anonymous functions can access variables not in its scope and it should work - https://go.dev/play/p/_GIdpQ89Lsp

Here you can access `backupNames` even without passing it to the function.

Reply:
I get your point that it's a better way and initially in my first commit was trying to do the same but it still fails. 

Dashboard URL : http://aetos.pwx.purestorage.com/resultSet/testSetID/121988

Reply:
Thank @mkoppal-px done and removed the loop